### ISSUE TITLE
Environment variable for `uv venv seed`?

### ISSUE THREAD

**Issue by ReubenFrankel** [astral-sh/uv#7655]
Hi! I'm using a tool called [Meltano](https://docs.meltano.com/) which supports `uv` as an alternate virtual environment backend. In our case, I would be great if `pip` was still bundled with the venvs `uv` creates for legacy reasons, which I can see is possible with the `uv venv --seed`. Since we don't have direct control over how `uv venv` runs behind the scenes, would it be possible to expose configuration for this via an environment variable (e.g. `UV_VENV_SEED=true`)?

Worth noting that I was running into #6593 initially - here is part of the message I sent in their Slack:

> I'm trying out `uv` as a venv backend, and it has been great up until running the `notebook` utility. Some of our notebooks contain an initial cell to `pip install` dependencies to the venv - with `uv`, this no longer works properly because `pip`  isn't bundled with the venv, so dependencies are installed to the system Python instead and then cannot be imported later on in the context of the venv Python.

**Comment by zanieb** [comment]
I have no qualms with exposing this via an environment variable, thoug, since nobody else has asked for this, I would appreciate if you opened an issue with Meltano asking for this flag too? The environment seeding concept exists separately from uv.

**Comment by ReubenFrankel** [comment]
Yeah, no problem - appreciate it's a bit of a niche issue.

**Comment by edgarrmondragon** [comment]
FWIW, virtualenv does support `VIRTUALENV_NO_SEED` and `VIRTUALENV_WITHOUT_PIP` (seeding is the default).

**Comment by woutervh** [comment]
I was going to open a new issue, but I'll it here:


Starting from a pyproject.toml we can create the venv via "uv sync".

How can we seed this venv with pip, equivalent to "uv venv  --seed"?
This is to ease the transaction for end-users.

Maybe an env-variable  UV_VENV_SEED?

**Comment by ssbarnea** [comment]
I am trying to improve the current implementation of https://github.com/tox-dev/tox-uv but this bug seems to be a major source of incompatibility, as it makes the uv created venvs incomptible with python's own venv ones.

**Comment by charliermarsh** [comment]
By “bug”, do you mean, the inability to configure this via an environment variable? It’s already configurable on the command line.

**Comment by edgarrmondragon** [comment]
Are the Astral folks open to a contribution adding support for the env var?

**Comment by charliermarsh** [comment]
Seems fine to me.

**Comment by charliermarsh** [comment]
I can do it today.

**Comment by edgarrmondragon** [comment]
thanks @charliermarsh!

**Comment by ReubenFrankel** [comment]
Thanks! 😎

**Comment by 341383** [comment]
Thanks for this! I'm poc'ing uv for our enterprise stack and was ironically looking for a solution the day that this was shipped. I've had the chance to do some initial testing and did have a question:

This seems to apply only when the environment is explicitly created with the `uv venv` command. However, if cloning a repo containing a uv.lock and then proceeding to create the venv implicitly with `uv sync`, the `UV_VENV_SEED` env variable is not respected. This was also observed with `uvx`.

Is this intentional or would it be possible to have creation of all (explicit and implicit) venvs respect the `UV_VENV_SEED` envVar?

**Comment by charliermarsh** [comment]
Yes, it's intentional. Enforcing it "everywhere" would be a separate decision.
